### DISCUSSION TITLE
SpanRuler to indicate spans of arbitary length

### DISCUSSION THREAD

**Discussion by Gitclop** [explosion/spaCy#13373] [Help: Best practices]
I accidently posted the question in the prodigy forum, but i think this belongs more in the spacy forum: 

I am training Spancat-Singlelabel to extract error messages from human written reports.
I've annotated arround 1200 examples and so far the results are OK (F1 Score of 0.75).

There are many error messages that are almost allways indicated by the same phrases. For example "... validation error: <error_message>. ..."
<error_message> can be different in length (often multiple sentences) and tokens. My training examples include a lot of those cases (annotated is only the error_message, not the indicating phrase before) but spaCy is still missing a lot of error_messages.

So my idea was to use SpanRuler to indicate those relevant passages or prephrases in front of the error messages but i am not sure how to use it to mark a span of arbitary length after it.

Or is it not the right tool for the job?

**Comment by honnibal** [comment]
You probably want to just have a custom component with some rule-based logic to get you the final outputs. The `Doc` and `Token` APIs are pretty nice to work with in spaCy, and you can also use the rule-based matcher. So I would have a process like this:

1. Detect the indicative phrases statistically
2. Have rules which recover the spans you need, based on the position of the indicative spans
